.github/workflows/nightly.yml

```diff
@@ -0,0 +1,43 @@
+name: nightly
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+
+jobs:
+  build-and-publish-nightly:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: |
+          python -m pip install -U pip
+          python -m pip install build
+      - name: Build the sdist
+        run: python -m build --sdist .
+        env:
+          BUILD_PYMC_NIGHTLY: true
+      - name: Publish to PyPI
+        uses: pypa/gh-action-pypi-publish@v1.5.0
+        with:
+          user: __token__
+          password: ${{ secrets.nightly_pypi_secret }}
+  test-install-job:
+    needs: build-and-publish-nightly
+    runs-on: ubuntu-latest
+    steps:
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+    - name: Give PyPI a chance to update the index
+      run: sleep 240
+    - name: Install from PyPI
+      run: |
+        pip install pymc-nightly==${GITHUB_REF:11}
```

Comment:
`${GITHUB_REF:11}` is a substring starting at the 11th position of the [`GITHUB_REF` environment variable](https://docs.github.com/en/actions/learn-github-actions/environment-variables). Quoting the docs:
> The branch or tag ref that triggered the workflow run. For branches this is the format refs/heads/<branch_name>, for tags it is refs/tags/<tag_name>, and for pull requests it is refs/pull/<pr_number>/merge. This variable is only set if a branch or tag is available for the event type. For example, refs/heads/feature-branch-1.

For example, with a tag `v1.2.3` the `GITHUB_REF` is `refs/heads/feature-branch-1` and `${GITHUB_REF:11}` evaluates to `1.2.3`.

So you'll have to do something else to get the nightly build version here.

Reply:
An example of the version would be 4.0.0b2.dev20220221. I've made an attempt following [this post](https://stackoverflow.com/questions/60942067/get-current-date-and-time-in-github-workflows), but it's difficult to test .yml. I skipped the step to create a date variable because we don't need to reuse it elsewhere in the workflow and the extra step makes it less readable – although I haven't tested whether it is _essential_ to create a variable for this to work

Reply:
The current version won't work, because the `GITHUB_REF` only includes the version number because it's a manually created Tag.
You'll need an environment variable of the current PyMC version, for example by extracting it from `__init__.py` with grep.

Does this have implications for _when_ we should bump our version number? Currently we do that right before a release.
Does the 4.0.0b2 version number even need to be included here?

Reply:
In the edited setup.py, I generate the pymc-nightly version using `suffix = datetime.now(timezone.utc).strftime(r".dev%Y%m%d")` and `version = f"{version}{suffix}"`.

The code showing on this thread is outdated, and I have updated to `pymc-nightly==${GITHUB_REF:11}.dev$(date +"%Y%m%d")`, which I hope translates to (e.g.) `pymc-nightly==4.0.0b2.dev20220221`.

I would say the version number is needed here. If we just ran `pip install pymc-nightly`, we cannot be sure we are downloading the _latest_ version of pymc-nightly.

To clarify, is the version number from `GITHUB_REF` 4.0.0 or 4.0.0b2?  If it's a manually created tag, all the nightly builds in between will still start with this version number, so as long as `GITHUB_REF` is updated each release, this should be fine